jans-auth-server/server/src/main/java/io/jans/as/server/authorize/ws/rs/AuthorizeRestWebServiceImpl.java

```diff
@@ -406,6 +406,14 @@ private Response requestAuthorization(
                 } catch (WebApplicationException e) {
                     JsonWebResponse jwr = parseRequestToJwr(request);
                     if (jwr != null) {
+                        String checkForAlg = jwr.getClaims().getClaimAsString("alg"); // to handle Jans Issue#310
+                        if (checkForAlg.equals("none")) {
```

Comment:
`"none".equals(alg)` - more defensive, it should not be null but claim itself is nullable.

Will `jwr.getClaims().getClaimAsString("alg")` gets inner request object `alg`, it looks like outer `alg` ?

Reply:
Thanks, I will change the condition as suggested.
During debugging, I checked and got the claims value as {alg:none} and this is why we don't get responseMode and state from claims.

Reply:
![Screenshot from 2022-02-09 15-33-37](https://user-images.githubusercontent.com/70174684/153176015-17ef37e3-251a-4902-86f8-1a8c90601c13.png)